### PR TITLE
feat: add new toggle component to dev overlay reference

### PR DESCRIPTION
**Pull request by Princesseuh**
#### Description (required)

https://github.com/withastro/astro/pull/9058 added a new toggle element checkbox component. This adds documentation for that! There's also a few tweaks that have been missed previously (more icons, an event name.. etc)

### CODE REVIEW

**Review by dreyfus92 — APPROVED** [review]
I'm so hyped about this, LGTM ✅

**Review by sarah11918 — APPROVED** [review]
LGTM! If this isn't waiting for the next minor, then feel free to release whenever the feature goes live!

**Pull request merged**
